src/main/java/com/exactprosystems/remotehand/web/actions/StoreElementState.java

```diff
@@ -59,17 +57,12 @@ public boolean isCanWait()
 	}
 	
 	@Override
-	public String run(WebDriver webDriver, By webLocator, Map<String, String> params) throws ScriptExecuteException
-	{
+	public String run(WebDriver webDriver, By webLocator, Map<String, String> params) throws ScriptExecuteException {
 		WebElement element = findElement(webDriver, webLocator);
-		byte[] screen = ScreenshotRegionUtils.takeElementScreenshot(webDriver, element);
-		Path fileName = Paths.get(WebConfiguration.SCREENSHOTS_DIR_NAME, "takeScreenshotAction" + System.currentTimeMillis() + ".jpeg");
-		try {
-			Files.write(fileName, screen);
-		} catch (IOException e) {
-			throw new ScriptExecuteException("Cannot store file", e);
-		}
-		context.getContextData().put(buildScreenshotId(params.get(PARAM_ID)), fileName);
+		String fileName = screenWriter.takeAndSaveElementScreenshot(screenshotName, webDriver, element);
```

Comment:
The same as wait for changes.
This action is used to create initial state on an element